qir/qat/ValidationPass/ValidationPassConfiguration.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+#include "QatTypes/QatTypes.hpp"
+#include "ValidationPass/ValidationPassConfiguration.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    void ValidationPassConfiguration::setup(ConfigurationManager& config)
+    {
+        config.setSectionName("Validation configuration", "");
+        config.addParameter(
+            allow_internal_calls_, "allow-internal-calls", "Whether or not internal calls are allowed.");
+        config.addParameter(
+            save_report_to_, "save-validation-report", "Saves the validation report to specified filename.");
+    }
+
+    String ValidationPassConfiguration::profileName() const
+    {
+        return profile_name_;
+    }
+
+    ValidationPassConfiguration ValidationPassConfiguration::fromProfileName(String const& name)
+    {
+
+        auto profile = ValidationPassConfiguration();
+        if (name == "generic")
+        {
+            profile.opcodes_               = Set{};
+            profile.external_calls_        = Set{};
+            profile.allowed_pointer_types_ = Set{};
+
+            profile.allowlist_opcodes_        = true;
+            profile.allowlist_external_calls_ = false;
+            profile.allow_internal_calls_     = true;
+            profile.allowlist_pointer_types_  = false;
+            profile.allow_primitive_return_   = true;
+            profile.allow_struct_return_      = true;
+            profile.allow_pointer_return_     = true;
+        }
+        else if (name == "base")
+        {
+            profile.allow_internal_calls_     = false;
+            profile.allowlist_external_calls_ = true;
+            profile.allowlist_opcodes_        = true;
+            profile.opcodes_        = Set{"br", "call", "unreachable", "ret", "inttoptr"}; //  "phi", "select",
```

Comment:
We can get into the details of refining the base profile later, I'm really focusing on the infrastructure for this review. Once we have an actual base profile definition, then we can adjust this list and consider removing the limited external calls below.

Reply:
Yes, I agree. This is not meant to be the final version, but it will be revised as we go.